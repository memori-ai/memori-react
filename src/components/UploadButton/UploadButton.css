/* Core Container */
.memori--unified-upload-wrapper {
  position: relative;
  display: inline-block;
  margin-right: 0.5rem;
}

/* Component Containers - Combined selectors for shared properties */
.memori--document-upload-wrapper,
.memori--image-upload-wrapper {
  position: relative;
  display: inline-block;
  margin-right: 0.5rem;
}

/* Hidden File Inputs */
.memori--upload-file-input {
  display: none;
}

/* Hidden Uploader Container - Using more accessible approach */
.memori--hidden-uploader {
  position: absolute;
  overflow: hidden;
  width: 1px;
  height: 1px;
  padding: 0;
  border: 0;
  margin: -1px;
  clip: rect(0, 0, 0, 0);
  white-space: nowrap;
}

/* Base button styles */
.memori--base-upload-button {
  position: relative;
  display: flex;
  width: 40px;
  height: 40px;
  align-items: center;
  justify-content: center;
  border: 1px solid #e0e0e0;
  border-radius: 50%;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
  cursor: pointer;
  transition: all 0.2s ease;
}

.memori--base-upload-button:hover {
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
  transform: translateY(-1px);
}

.memori--base-upload-button:active {
  box-shadow: 0 2px 3px rgba(0, 0, 0, 0.1);
  transform: translateY(0);
}

.memori--base-upload-button.memori--error {
  border-color: #ff4d4f;
  animation: pulse 1.5s infinite;
}

/* Main Upload Button */
.memori--unified-upload-button {
  position: relative;
  display: flex;
  width: 40px;
  height: 40px;
  align-items: center;
  justify-content: center;
  border: 1px solid #e0e0e0;
  border-radius: 50%;
  background-color: #f5f5f5;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
  cursor: pointer;
  transition: all 0.2s ease;
}

.memori--unified-upload-button:hover {
  background-color: #efefef;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
  transform: translateY(-1px);
}

/* Document Button */
.memori--document-upload-button {
  position: relative;
  display: flex;
  width: 40px;
  height: 40px;
  align-items: center;
  justify-content: center;
  border: 1px solid #fd7e14;
  border-radius: 50%;
  background-color: #fff9db;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
  cursor: pointer;
  transition: all 0.2s ease;
}

.memori--document-upload-button:hover {
  background-color: #fff3bf;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
  transform: translateY(-1px);
}

/* Image Button */
.memori--image-upload-button {
  position: relative;
  display: flex;
  width: 40px;
  height: 40px;
  align-items: center;
  justify-content: center;
  border: 1px solid #4dabf7;
  border-radius: 50%;
  background-color: #e7f5ff;
  box-shadow: 0 2px 5px rgba(0, 0, 0, 0.1);
  cursor: pointer;
  transition: all 0.2s ease;
}

.memori--image-upload-button:hover {
  background-color: #d0ebff;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
  transform: translateY(-1px);
}

/* Disabled button - Simplified */
.memori--image-upload-button:disabled {
  box-shadow: none;
  cursor: not-allowed;
  opacity: 0.5;
  transform: none;
}

/* Pulse animation */
@keyframes pulse {
  0% {
    box-shadow: 0 0 0 0 rgba(255, 77, 79, 0.4);
  }
  70% {
    box-shadow: 0 0 0 10px rgba(255, 77, 79, 0);
  }
  100% {
    box-shadow: 0 0 0 0 rgba(255, 77, 79, 0);
  }
}

/* Upload Icon */
.memori--upload-icon {
  width: 18px;
  height: 18px;
  color: #666;
}

@keyframes fadeIn {
  from {
    opacity: 0;
    transform: scale(0.95);
  }
  to {
    opacity: 1;
    transform: scale(1);
  }
}

/* Menu Items - Base styling */
.memori--upload-menu-item {
  display: flex;
  align-items: center;
  padding: 12px 16px;
  border-radius: 8px;
  margin: 4px 8px;
  cursor: pointer;
  transition: all 0.2s;
}

.memori--upload-menu-item:hover {
  background-color: #f5f5f5;
  transform: translateX(4px);
}


/* Menu icons - Consolidated styles */
.memori--upload-menu-icon,
.memori--upload-menu-icon-image {
  margin-right: 12px;
  color: var(--memori-primary, #4dabf7);
}

.memori--upload-menu-icon {
  width: 24px;
  height: 24px;
}

.memori--upload-menu-icon-image {
  width: 20px;
  height: 20px;
}

/* Preview Container */
.memori--preview-container {
  min-width: 200px;
  padding: 12px;
  border-radius: 8px;
  margin-bottom: 12px;
  animation: slide-in 0.3s ease;
  background: #fff;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1);
  transition: all 0.3s ease;
}

.memori--absolute-preview {
  position: absolute;
  z-index: 100;
  right: 0;
  bottom: 50px;
  left: auto;
  width: 300px;
  max-height: 300px;
  overflow-y: auto;
}

/* Preview Item - Base styling */
.memori--preview-item {
  position: relative;
  display: flex;
  align-items: center;
  padding: 10px 12px;
  border-radius: 8px;
  border-left: 3px solid transparent;
  margin-bottom: 8px;
  background-color: #f8f9fa;
  cursor: pointer;
  gap: 12px;
  transition: all 0.2s ease;
}

.memori--preview-item:hover {
  background-color: #f1f3f5;
  transform: translateX(4px);
}

/* Preview item variants */
.memori--preview-item--document {
  border-left-color: #fd7e14;
  background-color: #fff9db;
}

.memori--preview-item--image {
  border-left-color: #4dabf7;
  background-color: #e7f5ff;
}

.memori--preview-item--uploaded {
  border-left-color: #40c057;
}

.memori--preview-item--error {
  border-left-color: #fa5252;
}

/* Preview Icon */
.memori--preview-icon {
  width: 20px;
  height: 20px;
  flex-shrink: 0;
  fill: var(--memori-primary, #4dabf7);
}

/* Preview Thumbnail */
.memori--preview-thumbnail {
  overflow: hidden;
  width: 36px;
  height: 36px;
  flex-shrink: 0;
  border-radius: 4px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.memori--preview-thumbnail img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

/* Preview File Info */
.memori--preview-file-info {
  display: flex;
  overflow: hidden;
  flex-direction: column;
  flex-grow: 1;
}

.memori--preview-filename {
  overflow: hidden;
  max-width: 200px;
  color: #495057;
  font-size: 14px;
  font-weight: 500;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.memori--preview-filetype {
  display: flex;
  align-items: center;
  padding: 2px 0;
  margin-top: 2px;
  color: #868e96;
  font-size: 12px;
}

/* Upload Status */
.memori--upload-status {
  margin-left: 6px;
  color: #40c057;
  font-weight: 500;
}

.memori--upload-status-error {
  color: #fa5252;
}

/* Remove Button */
.memori--remove-button {
  position: absolute !important;
  top: -4px;
  right: -4px;
  width: 20px !important;
  height: 20px !important;
  padding: 4px;
  margin: 0;
  opacity: 0;
  transform: scale(0.75);
  transition: all 0.2s ease;
}

.memori--remove-button.visible {
  opacity: 1;
  transform: scale(1);
}

.memori--remove-button:hover {
  transform: scale(1.1) !important;
}

/* Modal Preview Content */
.memori--preview-content {
  padding: 16px;
  border-radius: 8px;
  margin: 0;
  background-color: #f8f9fa;
  font-family: monospace;
  line-height: 1.5;
}

.memori--modal-preview-file .memori--asset-info {
  padding: 12px;
  border: 1px solid #b2f2bb;
  border-radius: 8px;
  margin-top: 16px;
  background-color: #ebfbee;
  text-align: left;
}

.memori--asset-url {
  overflow: hidden;
  padding: 8px 12px;
  border-radius: 4px;
  margin-top: 8px;
  background-color: #f5f5f5;
  font-family: monospace;
  text-overflow: ellipsis;
  white-space: nowrap;
}

/* Positioned containers */
.memori--error-message-container,
.memori--login-tip {
  position: absolute;
  z-index: 1001;
  right: 0;
  bottom: 50px;
}

/* File Stats */
.memori--file-stats {
  position: absolute;
  z-index: 10;
  right: 0;
  bottom: -35px;
  padding: 4px 8px;
  border-radius: 4px;
  margin: 0;
  background-color: #f5f5f5;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  color: #666;
  font-size: 12px;
}

.memori--file-stats p {
  padding: 2px 0;
  margin: 0;
}

/* Improved image count indicator */
.memori--image-count {
  position: absolute;
  z-index: 5;
  top: -8px;
  right: -8px;
  padding: 1px 6px;
  border-radius: 10px;
  background-color: #4dabf7;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.2);
  color: white;
  font-size: 11px;
  font-weight: bold;
}

.memori--image-count-full {
  background-color: #fa5252;
}

/* Styled upload slots info */
.memori--upload-slots-info {
  margin-left: 4px;
  color: #868e96;
  font-size: 12px;
  font-style: italic;
}

/* Disabled menu item styling */
.memori--upload-menu-item--disabled {
  cursor: not-allowed;
  opacity: 0.5;
}

.memori--upload-menu-item--disabled:hover {
  background-color: transparent;
  transform: none;
}

/* Enhanced floating menu */
.memori--upload-menu {
  position: absolute;
  z-index: 1000;
  right: 0;
  bottom: 50px;
  width: 220px; /* Slightly wider to accommodate the text */
  padding: 8px 0;
  border-radius: 12px;
  animation: fadeIn 0.2s ease;
  background-color: #fff;
  box-shadow: 0 8px 24px rgba(0, 0, 0, 0.15);
  transform-origin: bottom right;
}

/* Menu item hover effect */
.memori--upload-menu-item:hover:not(.memori--upload-menu-item--disabled) {
  background-color: #f5f5f5;
  transform: translateX(4px);
}

/* Error container positioning */
.memori--error-message-container {
  position: absolute;
  z-index: 1001;
  right: 0;
  bottom: 50px;
  min-width: 250px;
}

input.memori--upload-title-input {
  min-height: 2.5rem;
  border: 1px solid #e0e0e0;
  border-radius: 0.375rem;
  appearance: none;
  background: inherit;
  font-size: 1rem;
  outline: transparent solid 2px;
  outline-offset: 2px;
  padding-inline-end: 1rem;
  padding-inline-start: 1rem;
  transition-duration: 0.2s;
  transition-property: background-color, border-color, color, fill, stroke, opacity, box-shadow, transform;
}

/* Animation Keyframes */
@keyframes slide-in {
  from {
    opacity: 0;
    transform: translateY(-10px);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
}

@keyframes slide-down {
  from {
    opacity: 1;
    transform: translateY(0);
  }
  to {
    opacity: 0;
    transform: translateY(10px);
  }
}

/* Responsive adjustments */
@media (max-width: 768px) {
  .memori--absolute-preview {
    width: 250px;
  }

  .memori--upload-menu {
    width: 180px;
  }
}