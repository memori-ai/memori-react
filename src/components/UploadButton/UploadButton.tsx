import React, { useState, useRef, useEffect } from 'react';
import { DocumentIcon } from '../icons/Document';
import { ImageIcon } from '../icons/Image';
import { UploadIcon } from '../icons/Upload';
import Spin from '../ui/Spin';
import Alert from '../ui/Alert';
import cx from 'classnames';
import UploadDocuments from './UploadDocuments/UploadDocuments';
import UploadImages from './UploadImages/UploadImages';
import { useTranslation } from 'react-i18next';

// Constants
const MAX_IMAGES = 5;
const MAX_DOCUMENTS = 1;

// Props interface
interface UploadManagerProps {
  authToken?: string;
  apiUrl?: string;
  sessionID?: string;
  isMediaAccepted?: boolean;
  setDocumentPreviewFiles: any;
  documentPreviewFiles: {
    name: string;
    id: string;
    content: string;
    mediumID?: string;
    type?: string;
  }[];
}

const UploadButton: React.FC<UploadManagerProps> = ({
  authToken = '',
  apiUrl = '',
  sessionID = '',
  isMediaAccepted = false,
  setDocumentPreviewFiles,
  documentPreviewFiles,
}) => {
  // State
  const [isLoading, setIsLoading] = useState(false);
  const [menuOpen, setMenuOpen] = useState(false);
  const [errors, setErrors] = useState<
    { message: string; severity: 'error' | 'warning' | 'info' }[]
  >([]);
  const { t, i18n } = useTranslation();

  // Refs
  const menuRef = useRef<HTMLDivElement>(null);
  const buttonRef = useRef<HTMLButtonElement>(null);
  const documentRef = useRef<HTMLDivElement>(null);
  const imageRef = useRef<HTMLDivElement>(null);

  // Calculate image count and remaining slots
  const currentImageCount = documentPreviewFiles.filter(file => file.type === 'image').length;
  const remainingSlots = MAX_IMAGES - currentImageCount;
  const currentDocumentCount = documentPreviewFiles.filter(file => file.type === 'document').length;
  const remainingDocumentSlots = MAX_DOCUMENTS - currentDocumentCount;
  const hasReachedImageLimit = remainingSlots <= 0;
  const hasReachedDocumentLimit = remainingDocumentSlots <= 0;

  // Error handling
  const clearErrors = () => setErrors([]);

  const removeError = (errorMessage: string) => {
    setErrors(prev => prev.filter(e => e.message !== errorMessage));
  };

  const addError = (error: {
    message: string;
    severity: 'error' | 'warning' | 'info';
  }) => {
    setErrors(prev => [...prev, error]);
    setTimeout(() => removeError(error.message), 5000);
  };

  // Menu handling
  const toggleMenu = () => {
    setMenuOpen(prev => !prev);
  };

  const closeMenu = () => {
    setMenuOpen(false);
  };

  // Click outside handler
  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (
        menuRef.current &&
        buttonRef.current &&
        !menuRef.current.contains(event.target as Node) &&
        !buttonRef.current.contains(event.target as Node)
      ) {
        closeMenu();
      }
    };

    document.addEventListener('mousedown', handleClickOutside);
    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
    };
  }, []);

  // Handler for document files - only stores the latest document
  const handleDocumentFiles = (
    files: { name: string; id: string; content: string }[]
  ) => {
    if (files.length === 0) return;

    // For simplicity, we only take the first file
    const file = files[0];

    // Format content with XML tags to improve readability for LLM
    const formattedContent = `<Documento allegato al messaggio: ${file.name}>
${file.content}
</Documento allegato al messaggio: ${file.name}>`;

    //keep just the images in the documentPreviewFiles
    const imageFiles = documentPreviewFiles.filter(
      (file: any) => file.type === 'image'
    );

    // Replace existing file with new one
    setDocumentPreviewFiles([
      {
        name: file.name,
        id: file.id,
        content: formattedContent,
        type: 'document'
      },
      ...imageFiles,
    ]);

    setIsLoading(false);
  };

  // When document option is clicked
  const handleDocumentClick = () => {
    // Find the actual button in the UploadDocuments component and click it
    const documentButtonElement = documentRef.current?.querySelector('button');
    if (documentButtonElement) {
      documentButtonElement.click();
    }
    closeMenu();
  };

  // When image option is clicked
  const handleImageClick = () => {
    if (!authToken) {
      addError({
        message: t('upload.loginRequired') ?? 'Login required to upload images',
        severity: 'info',
      });
      closeMenu();
      return;
    }
    
    if (!isMediaAccepted) {
      addError({
        message: t('upload.mediaNotAccepted') ?? 'Media uploads are not accepted',
        severity: 'info',
      });
      closeMenu();
      return;
    }
    
    if (hasReachedImageLimit) {
      addError({
        message: t('upload.maxImagesReached', { max: MAX_IMAGES }) ?? 
          `Maximum ${MAX_IMAGES} images already uploaded`,
        severity: 'warning',
      });
      closeMenu();
      return;
    }
    
    // If all checks pass, click the button in UploadImages component
    const imageButtonElement = imageRef.current?.querySelector('button');
    if (imageButtonElement) {
      imageButtonElement.click();
    }
    closeMenu();
  };

  // Set loading state for child components
  const handleLoadingChange = (loading: boolean) => {
    setIsLoading(loading);
  };

  return (
    <div className="memori--unified-upload-wrapper">
      {/* Main upload button */}
      <button
        ref={buttonRef}
        className={cx(
          'memori-button',
          'memori-button--circle',
          'memori-button--icon-only',
          'memori-share-button--button',
          'memori--conversation-button',
          'memori--unified-upload-button',
          { 'memori--error': errors.length > 0 }
        )}
        onClick={toggleMenu}
        disabled={isLoading}
        title={t('upload.uploadFiles') ?? 'Upload files'}
      >
        {isLoading ? (
          <Spin spinning className="memori--upload-icon" />
        ) : (
          <UploadIcon className="memori--upload-icon" />
        )}
      </button>
      
      {/* Image count indicator - moved here from UploadImages */}
      {currentImageCount > 0 && (
        <div className={cx(
          'memori--image-count',
          { 'memori--image-count-full': hasReachedImageLimit }
        )}>
          {currentImageCount}/{MAX_IMAGES}
        </div>
      )}

      {/* Floating menu */}
      {menuOpen && (
        <div className="memori--upload-menu" ref={menuRef}>
          <div
             className={cx('memori--upload-menu-item', {
              'memori--upload-menu-item--disabled':
                !authToken || hasReachedDocumentLimit,
            })}
            onClick={handleDocumentClick}
          >
            <DocumentIcon className="memori--upload-menu-icon" />
            <span>
              {t('upload.uploadDocument') ?? 'Upload document'}
              {currentDocumentCount > 0 && (
                <span className="memori--upload-slots-info">
                  {hasReachedDocumentLimit
                    ? ` (${t('upload.maxReached') ?? 'Max reached'})`
                    : ` (${remainingDocumentSlots} ${t('upload.remaining') ?? 'remaining'})`
                  }
                </span>
              )}
            </span>
          </div>

          <div
            className={cx('memori--upload-menu-item', {
              'memori--upload-menu-item--disabled':
                !isMediaAccepted || !authToken || hasReachedImageLimit,
            })}
            onClick={handleImageClick}
            title={
              !authToken
                ? t('upload.loginRequired') ?? 'Login Required'
                : !isMediaAccepted
                  ? t('upload.mediaNotAccepted') ?? 'Media uploads not accepted'
                  : hasReachedImageLimit
                    ? t('upload.maxImagesReached', { max: MAX_IMAGES }) ?? `Maximum ${MAX_IMAGES} images already uploaded`
                    : remainingSlots === 1
                      ? t('upload.lastImageSlot') ?? 'Upload last image'
                      : t('upload.uploadImage', { remaining: remainingSlots }) ?? `Upload image (${remainingSlots} remaining)`
            }
          >
            <ImageIcon className="memori--upload-menu-icon-image" />
            <span>
              {t('upload.uploadImage') ?? 'Upload image'}
              {currentImageCount > 0 && (
                <span className="memori--upload-slots-info">
                  {hasReachedImageLimit
                    ? ` (${t('upload.maxReached') ?? 'Max reached'})`
                    : ` (${remainingSlots} ${t('upload.remaining') ?? 'remaining'})`
                  }
                </span>
              )}
            </span>
          </div>
        </div>
      )}

      {/* Hidden components */}
      <div className="memori--hidden-uploader" ref={documentRef}>
        <UploadDocuments 
          setDocumentPreviewFiles={handleDocumentFiles}
          maxDocuments={MAX_DOCUMENTS}
          documentPreviewFiles={documentPreviewFiles}
          // onLoadingChange={handleLoadingChange}
        />
      </div>

      <div className="memori--hidden-uploader" ref={imageRef}>
        <UploadImages
          authToken={authToken}
          apiUrl={apiUrl}
          setDocumentPreviewFiles={setDocumentPreviewFiles}
          sessionID={sessionID}
          documentPreviewFiles={documentPreviewFiles}
          isMediaAccepted={isMediaAccepted}
          onLoadingChange={handleLoadingChange}
          // Pass the constants to UploadImages
          maxImages={MAX_IMAGES}
        />
      </div>

      {/* Error messages container */}
      <div className="memori--error-message-container">
        {errors.map((error, index) => (
          <Alert
            key={`${error.message}-${index}`}
            open={true}
            type={error.severity}
            title={'Upload notification'}
            description={error.message}
            onClose={() => removeError(error.message)}
            width="350px"
          />
        ))}
      </div>

      {/* Login tip */}
      {!authToken && menuOpen && (
        <div className="memori--login-tip">
          <Alert
            type="info"
            title={t('upload.loginRequired') ?? 'Login Required'}
            description={t('upload.loginRequiredDescription') ?? 'Please login to upload images'}
            width="350px"
            onClose={closeMenu}
          />
        </div>
      )}
    </div>
  );
};

export default UploadButton;