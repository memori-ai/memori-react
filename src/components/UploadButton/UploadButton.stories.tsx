import React, { useState } from 'react';
import { Meta, Story } from '@storybook/react';
import I18nWrapper from '../../I18nWrapper';
import UploadButton from './UploadButton';
import FilePreview from '../FilePreview/FilePreview';
import './UploadButton.css';

// Define the component props
interface UploadButtonProps {
  authToken?: string;
  apiUrl?: string;
  sessionID?: string;
  isMediaAccepted?: boolean;
  setDocumentPreviewFiles: (files: { name: string; id: string; content: string; type?: string }[]) => void;
  disabled?: boolean;
}

const meta: Meta<UploadButtonProps> = {
  title: 'Widget/Upload Button',
  component: UploadButton,
  argTypes: {
    disabled: {
      control: {
        type: 'boolean',
      },
    },
    isMediaAccepted: {
      control: {
        type: 'boolean',
      },
    },
    authToken: {
      control: {
        type: 'text',
      },
    },
    apiUrl: {
      control: {
        type: 'text',
      },
    },
    sessionID: {
      control: {
        type: 'text',
      },
    },
  },
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

// Create the template
const Template: Story<UploadButtonProps> = (args) => {
  const [documentPreviewFiles, setDocumentPreviewFiles] = useState<{
    name: string;
    id: string;
    content: string;
    type?: string;
  }[]>([]);

  const removeFile = (id: string) => {
    setDocumentPreviewFiles(prev => prev.filter(file => file.id !== id));
  };

  return (
    <I18nWrapper>
      <div
        style={{
          minHeight: '400px',
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'flex-start',
          justifyContent: 'center',
          padding: '20px',
          border: '1px solid #eee',
          borderRadius: '8px',
        }}
      >
        <div style={{ alignSelf: 'flex-end', marginBottom: '20px' }}>
          <UploadButton {...args} setDocumentPreviewFiles={setDocumentPreviewFiles} />
        </div>
        
        {documentPreviewFiles.length > 0 && (
          <div style={{ width: '100%', marginTop: '20px' }}>
            <h3>Uploaded Files</h3>
            <FilePreview 
              previewFiles={documentPreviewFiles} 
              removeFile={removeFile}
            />
          </div>
        )}
      </div>
    </I18nWrapper>
  );
};

// Default state - no auth token, documents only
export const Default = Template.bind({});
Default.args = {
  authToken: '',
  apiUrl: 'https://api.example.com',
  sessionID: 'session-123',
  isMediaAccepted: false,
};

// Authenticated user - can upload documents
export const AuthenticatedDocumentsOnly = Template.bind({});
AuthenticatedDocumentsOnly.args = {
  authToken: 'dummy-auth-token-12345',
  apiUrl: 'https://api.example.com',
  sessionID: 'session-123',
  isMediaAccepted: false,
};

// Authenticated user - can upload documents and images
export const AuthenticatedWithImages = Template.bind({});
AuthenticatedWithImages.args = {
  authToken: 'dummy-auth-token-12345',
  apiUrl: 'https://api.example.com',
  sessionID: 'session-123',
  isMediaAccepted: true,
};

// Not authenticated but media accepted (should show warning for images)
export const UnauthenticatedWithMediaAccepted = Template.bind({});
UnauthenticatedWithMediaAccepted.args = {
  authToken: '',
  apiUrl: 'https://api.example.com',
  sessionID: 'session-123',
  isMediaAccepted: true,
};

// Disabled state
export const Disabled = Template.bind({});
Disabled.args = {
  authToken: 'dummy-auth-token-12345',
  apiUrl: 'https://api.example.com',
  sessionID: 'session-123',
  isMediaAccepted: true,
  disabled: true,
};


// Mobile view simulation
export const MobileView = Template.bind({});
MobileView.args = {
  authToken: 'dummy-auth-token-12345',
  apiUrl: 'https://api.example.com',
  sessionID: 'session-123',
  isMediaAccepted: true,
};
MobileView.parameters = {
  viewport: {
    defaultViewport: 'mobile1',
  },
  docs: {
    description: {
      story: 'A simulation of how the upload button appears on mobile devices.',
    },
  },
};