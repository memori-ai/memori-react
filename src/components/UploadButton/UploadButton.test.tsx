import React from 'react';
import { render } from '@testing-library/react';
import UploadButton from './UploadButton';

it('renders UploadButton unchanged', () => {
  const { container } = render(
    <UploadButton setDocumentPreviewFiles={jest.fn()} documentPreviewFiles={[]} />
  );
  expect(container).toMatchSnapshot();
});

