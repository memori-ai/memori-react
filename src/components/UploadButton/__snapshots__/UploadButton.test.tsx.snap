// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UploadButton unchanged 1`] = `
<div>
  <div
    class="memori--unified-upload-wrapper"
  >
    <button
      class="memori-button memori-button--circle memori-button--icon-only memori-share-button--button memori--conversation-button memori--unified-upload-button"
      title="Upload files"
    >
      <svg
        class="memori--upload-icon"
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 15V19C21 19.5304 20.7893 20.0391 20.4142 20.4142C20.0391 20.7893 19.5304 21 19 21H5C4.46957 21 3.96086 20.7893 3.58579 20.4142C3.21071 20.0391 3 19.5304 3 19V15"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M17 8L12 3L7 8"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
        <path
          d="M12 3V15"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </button>
    <div
      class="memori--hidden-uploader"
    >
      <div
        class="memori--document-upload-wrapper"
      >
        <input
          accept=".pdf,.txt,.md,.json,.xlsx,.csv"
          class="memori--upload-file-input"
          type="file"
        />
        <button
          class="memori-button memori-button--circle memori-button--icon-only memori-share-button--button memori--conversation-button memori--document-upload-button"
          title="Upload documents"
        >
          <svg
            class="memori--upload-icon"
            fill="none"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 2.5H6C5.60218 2.5 5.22064 2.65804 4.93934 2.93934C4.65804 3.22064 4.5 3.60218 4.5 4V20C4.5 20.3978 4.65804 20.7794 4.93934 21.0607C5.22064 21.342 5.60218 21.5 6 21.5H18C18.3978 21.5 18.7794 21.342 19.0607 21.0607C19.342 20.7794 19.5 20.3978 19.5 20V8L14 2.5Z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M14 2.5V8H19.5"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M16 13H8"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M16 17H8"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M10 9H8"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </button>
        <div
          class="memori--error-message-container"
        />
      </div>
    </div>
    <div
      class="memori--hidden-uploader"
    >
      <div
        class="memori--image-upload-wrapper"
      >
        <input
          accept=".jpg,.jpeg,.png"
          class="memori--upload-file-input"
          disabled=""
          multiple=""
          type="file"
        />
        <button
          class="memori-button memori-button--circle memori-button--icon-only memori-share-button--button memori--conversation-button memori--image-upload-button"
          disabled=""
          title="Please login to upload images"
        >
          <svg
            class="memori--upload-icon"
            fill="none"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="18"
              rx="2"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              width="18"
              x="3"
              y="3"
            />
            <circle
              cx="8.5"
              cy="8.5"
              fill="currentColor"
              r="1.5"
            />
            <path
              d="M21 15L16 10L5 21"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </button>
        <div
          class="memori--error-message-container"
        />
        <div
          class="memori--login-tip"
        />
      </div>
    </div>
    <div
      class="memori--error-message-container"
    />
  </div>
</div>
`;
